job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -183,6 +184,52 @@ BEGIN
          ELSE quote_ident(in_sort_field)
        END ||
         ' ' || CASE WHEN in_sort_ascending THEN 'ASC' ELSE 'DESC' END;
-    RETURN QUERY EXECUTE sql;
+
+    -- Create temporary table as a base to update the job progress
+    EXECUTE 'CREATE TEMPORARY TABLE get_job_temp ON COMMIT DROP AS ' || sql;
+
+    ALTER TABLE get_job_temp ADD COLUMN id SERIAL PRIMARY KEY;
+
+    -- Create a duplicate of that temporary table to store the results after the update
+    CREATE TEMPORARY TABLE new_table ON COMMIT DROP AS SELECT * FROM get_job_temp ORDER BY id;
+
+    -- Remove duplicates based on the label
+    DELETE FROM get_job_temp a
+    USING get_job_temp b
+    WHERE a.id < b.id
+        AND a.label = b.label;
```

Comment:
I get the concept of removing the duplicates but does this statement really do that?  It's a difficult one to understand.  Why don't you just change the statement below to `FOR jobId IN SELECT DISTINCT(jobId) FROM get_job_temp LOOP`?  I think it would be easier to understand.

Reply:
replaced the second temp table with a simplified array